greenpithumb/camera.py

```diff
@@ -0,0 +1,27 @@
+class Camera(object):
+    """Wrapper for a Raspberry Pi Camera."""
+
+    def __init__(self, folder_path, local_clock, camera):
+        """Creates a new camera wrapper instance.
+
+        Args:
+            folder_path: String of the path name of the folder where pictures
+                will be stored.
+            local_clock: Local clock interface.
+            camera: Camera interface.
+        """
+        self._folder_path = folder_path
+        self._local_clock = local_clock
+        self._camera = camera
+
+    def capture(self):
+        """Captures an image from the camera."""
+        filename = self._local_clock.now().strftime('%Y-%m-%d-%H')
+        self._camera.capture(self._folder_path + filename, format='jpeg')
```

Comment:
We always want to use [`os.path.join`](https://docs.python.org/2/library/os.path.html#os.path.join) to combine file paths. There's a lot of subtle things that can go wrong if we do it by hand (especially if we want to work well across platforms). A common bug is to call the function and forget the trailing slash like `C:\foo\bar` instead of `C:\foo\bar\` and then it puts the file at `C:\foo\bar2016-01-05.jpeg`.